packages/next-swc/crates/next-build/src/next_pages/page_entries.rs

```diff
@@ -308,16 +314,133 @@ pub struct PageEntry {
 async fn get_page_entry_for_file(
     ssr_module_context: AssetContextVc,
     client_module_context: AssetContextVc,
-    source: SourceVc,
     next_router_root: FileSystemPathVc,
-    next_router_path: FileSystemPathVc,
+    app: PagesStructureItemVc,
+    path_type: PathType,
 ) -> Result<PageEntryVc> {
-    let reference_type = Value::new(ReferenceType::Entry(EntryReferenceSubType::Page));
+    let app = app.await?;
+    let source: SourceVc = FileSourceVc::new(app.project_path).into();
+    let next_router_path = app.next_router_path;
+    let reference_type = match path_type {
+        PathType::Page => Value::new(ReferenceType::Entry(EntryReferenceSubType::Page)),
+        PathType::PagesAPI => Value::new(ReferenceType::Entry(EntryReferenceSubType::PagesApi)),
+        _ => bail!("Invalid path type"),
+    };
 
-    let pathname = pathname_for_path(next_router_root, next_router_path, PathType::Page);
+    let pathname = pathname_for_path(next_router_root, next_router_path, path_type);
+
+    let definition_page = format!("/{}", app.original_path.await?);
+    let definition_pathname = pathname.await?;
 
     let ssr_module = ssr_module_context.process(source, reference_type.clone());
 
+    let mut result = RopeBuilder::default();
+
+    match path_type {
+        PathType::Page => {
+            // Sourced from https://github.com/vercel/next.js/blob/2848ce51d1552633119c89ab49ff7fe2e4e91c91/packages/next/src/build/webpack/loaders/next-route-loader/index.ts
```

Comment:
Instead of copying the code for turbopack and we pack, you can put that code into a shared JS file.
This can be imported in turobpack, and a webpack can read the file and replace `"INNER"` with string replace

Reply:
If we put this into a JS file, we can't perform the following:

https://github.com/vercel/next.js/blob/ef8255dbf4a2e1b53be4eb1b5472d322be6d946d/packages/next-swc/crates/next-build/src/next_pages/page_entries.rs#L386-L394

Or

https://github.com/vercel/next.js/blob/503677c0998e3d0711359644d5d7caf3cc067491/packages/next/src/build/webpack/loaders/next-route-loader/index.ts#L129-L136

Reply:
Reading the file from disk will give us a `RopeVc`, which we can then use to build another rope that we can concatenate into.